Jack/Script.fsx

```diff
@@ -112,6 +112,30 @@ Property.print <| property {
 }
 
 //
+// Loops
+//
+
+Property.print <| property {
+    for x in "abcd" do
+        // Custom operations (i.e. counterexample) can't be used in computation
+        // expressions which have control flow :( we can fake it using return!
+        // however.
+        return! Property.counterexample (sprintf "x = %A" x)
+
+        // Note, return can be used multiple times, its a bit like 'assert'.
+        return x <> 'w'
+        return x <> 'z'
+
+    let mutable n = 0
+    while n < 10 do
+        n <- n + 1
+        let! k = Gen.range 0 n
+        return! Property.counterexample (sprintf "n = %d" n)
+        return! Property.counterexample (sprintf "k = %d" k)
+        return k <> 5
```

Comment:
This shows the crazy stuff which you can do with loops, I'm not sure if this is actually useful yet, we'll see.

It boggles the mind to think what is happening under the hood with the rose tree.